src/ControlSystem/Actions/InitializeMeasurements.hpp

```diff
@@ -119,6 +129,40 @@ struct InitializeMeasurements {
               });
         },
         make_not_null(&box));
+
+    // Ensure that the initial time step is small enough that we don't
+    // need to perform any measurements to complete it.  This is only
+    // necessary for TimeSteppers that use the self-start procedure,
+    // because they cannot adjust their step size on the first step
+    // after self-start, but it isn't harmful to do it in other cases.
+    //
+    // Unlike in the steady-state step-limiting code, we don't do
+    // anything clever looking at measurement times or planning ahead
+    // for future steps.  Avoiding a single non-ideal step isn't worth
+    // the added complexity.
+    double earliest_expiration = std::numeric_limits<double>::infinity();
+    for (const auto& fot :
+         Parallel::get<domain::Tags::FunctionsOfTime>(cache)) {
+      earliest_expiration =
+          std::min(earliest_expiration, fot.second->time_bounds()[1]);
+    }
+    const auto& time_step = db::get<::Tags::TimeStep>(box);
+    const auto start_time = time_step.slab().start();
+    if ((start_time + time_step).value() > earliest_expiration) {
+      TimeDelta new_step{};
+      if constexpr (Metavariables::local_time_stepping) {
```

Comment:
[optional] You could put the `if constexpr` inside the `db::mutate`. Avoids the intermediate variable.